### PR TITLE
fix: \\n should be changed to \n only when escapeSpecialCharacters is set

### LINKED ISSUE #90: 🐛 Bug: \\n -> \n should only happen with template strings
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/dmnd/dedent/issues?q=is%3Aissue) and found none that matched my issue.

### Expected

`dedent("\\nu")` should return the string unchanged.

### Actual

`dedent("\\nu")` turns the \\\n into a newline, i.e. returns a newline character + `u`.

### Additional Info

This happens because `result.replace(/\\n/g, "\n")` is called regardless of the value of `escapeSpecialCharacters`, i.e. regardless of whether dedent is called as a tag on a template or as a function on a string.
An easy fix would be to make the replacement conditional, solving [this particular problem](https://github.com/marimo-team/marimo/issues/1004). (However, it would still not work for ``dedent`\\nu` `` because here, still the \\\n would be replaced 🤷‍♂️)

### PR DESCRIPTION
**Pull request by fuenfundachtzig**
This PR proposes a fix to the issue #90.

## PR Checklist

- [x] Addresses an existing open issue: fixes #90
- [x] That issue was marked as [`status: accepting prs`](https://github.com/dmnd/dedent/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/dmnd/dedent/blob/main/.github/CONTRIBUTING.md) were taken

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
 	"name": "dedent",
-	"version": "1.5.1",
+	"version": "1.5.2",
```
Comment (JoshuaKGoldberg) [inline package.json:3]: No need 🙂 I'll manage this. If other PRs come in the version might not be 1.5.*.

```suggestion
	"version": "1.5.1",
```

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
LGTM! Could you add a test to `dedent-tests.ts` please?

**Comment by fuenfundachtzig** [comment]
> LGTM! Could you add a test to `dedent-tests.ts` please?

Thanks! I have added a test.

**Review by JoshuaKGoldberg — APPROVED** [review]
🔥 thanks!

**Pull request merged**